src/raw.imageio/rawinput.cpp

```diff
@@ -193,30 +200,35 @@ RawInput::open (const std::string &name, ImageSpec &newspec,
     }
 
     // Interpolation quality
-    // note: LibRaw must be compiled with demosaic pack GPL2 to use
-    // demosaic algorithms 5-9. It must be compiled with demosaic pack GPL3 for 
-    // algorithm 10. If either of these packs are not includeded, it will silently use option 3 - AHD
+    // note: LibRaw must be compiled with demosaic pack GPL2 to use demosaic
+    // algorithms 5-9. It must be compiled with demosaic pack GPL3 for
+    // algorithm 10 (AMAzE). If either of these packs are not includeded, it
+    // will silently use option 3 - AHD.
     std::string demosaic = config.get_string_attribute ("raw:Demosaic");
     if (demosaic.size()) {
         static const char *demosaic_algs[] = { "linear",
                                                "VNG",
                                                "PPG",
                                                "AHD",
                                                "DCB",
-                                               "Modified AHD",
+                                               "AHD-Mod",
                                                "AFD",
                                                "VCD",
                                                "Mixed",
                                                "LMMSE",
                                                "AMaZE",
+#if LIBRAW_VERSION >= LIBRAW_MAKE_VERSION(0,16,0)
+                                               "DHT",
+                                               "AAHD",
+#endif
                                                // Future demosaicing algorithms should go here
                                                NULL
                                                };
         size_t d;
         for (d=0; d < sizeof(demosaic_algs) / sizeof(demosaic_algs[0]); d++)
-            if (demosaic == demosaic_algs[d])
+            if (Strutil::iequals (demosaic, demosaic_algs[d]))
                 break;
-        if (demosaic == demosaic_algs[d])
+        if (Strutil::iequals (demosaic, demosaic_algs[d]))
```

Comment:
Actually, I think it can just be

    if (demosaic_algs[d])

since it will only be non-NULL if it matched something.